pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ClientCnx.java

```diff
@@ -61,6 +64,13 @@
 
     private final CompletableFuture<Void> connectionFuture = new CompletableFuture<Void>();
     private final Semaphore pendingLookupRequestSemaphore;
+    private final Timer timer;
```

Comment:
actually we don't have `ScheduledExecutor` instance around. should we create one?